From 23c2cdcda7cfb0c8609506ca7789b9a69eba11f7 Mon Sep 17 00:00:00 2001
From: Antoine du Hamel <duhamelantoine1995@gmail.com>
Date: Sun, 11 Oct 2026 14:55:49 +0200
Subject: [PATCH] deps: V8: cherry-pick 8d14d49c4040

Original commit message:

    fix compilation with GCC 16 on arm64

    Change-Id: I2cbe5f2a68d4a69bac48eea876ff0ed0cb589171
    Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/8500805
    Reviewed-by: Joyee Cheung <joyee@igalia.com>
    Commit-Queue: Joyee Cheung <joyee@igalia.com>
    Reviewed-by: Anton Bikineev <bikineev@chromium.org>
    Cr-Commit-Position: refs/heads/main@{#110517}

Refs: https://github.com/v8/v8/commit/8d14d49c40408981bed6a73dcdba9cd3e2e79a89
PR-URL: https://github.com/nodejs/node/pull/66627
Reviewed-By: Richard Lau <richard.lau@ibm.com>
Reviewed-By: Chengzhong Wu <legendecas@gmail.com>
Reviewed-By: Luigi Pinca <luigipinca@gmail.com>
Signed-off-by: stydxm <stydxm@outlook.com>
---
 common.gypi                | 2 +-
 deps/v8/src/base/memcopy.h | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/common.gypi b/common.gypi
index de80a68f492..eeebab06807 100644
--- a/common.gypi
+++ b/common.gypi
@@ -44,7 +44,7 @@
 
     # Reset this number to 0 on major V8 upgrades.
     # Increment by one for each non-official patch applied to deps/v8.
-    'v8_embedder_string': '-node.37',
+    'v8_embedder_string': '-node.38',
 
     ##### V8 defaults for Node.js #####
 
diff --git a/deps/v8/src/base/memcopy.h b/deps/v8/src/base/memcopy.h
index b1d8a36b4cf..8254318fe52 100644
--- a/deps/v8/src/base/memcopy.h
+++ b/deps/v8/src/base/memcopy.h
@@ -5,6 +5,7 @@
 #ifndef V8_BASE_MEMCOPY_H_
 #define V8_BASE_MEMCOPY_H_
 
+#include <limits.h>
 #include <stdlib.h>
 
 #include <atomic>
-- 
2.34.1

